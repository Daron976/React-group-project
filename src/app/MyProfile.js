import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const joinedMissions = useSelector((state) => state.missionReducer);
  const rocketsData = useSelector((state) => state.rocketReducer);

  return (
    <section className="profile-container">
      <section className="join-missions">
        <h1>My Missions</h1>
        <ul className="mission-list">
          {joinedMissions
            .filter((mission) => mission.reserved === true)
            .map((mission) => (
              <li className="mission-list-item" key={mission.mission_id}>
                {mission.mission_name}
              </li>
            ))}
        </ul>
      </section>
      <section className="reserved-rockets">
        <h1>My Rockets</h1>
        <ul className="rocket-list">
          {rocketsData
            .filter((rocket) => rocket.reserved === true)
            .map((rocket) => (
              <li className="rocket-list-item" key={rocket.id}>
                {rocket.rocket_name}
              </li>
            ))}
        </ul>
      </section>
    </section>
  );
};

export default MyProfile;
