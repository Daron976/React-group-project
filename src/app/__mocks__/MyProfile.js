import React from 'react';

const MyProfile = () => {
  const joinedMissions = [
    {
      id: 1,
      name: 'test',
      description: 'this is a test description',
      reserved: true,
    },
  ];

  const rocketsData = [
    {
      id: 1,
      name: 'test',
      description: 'this is a test description',
      reserved: true,
    },
  ];

  return (
    <section className="profile-container">
      <section className="join-missions">
        <h1>My Missions</h1>
        <ul className="mission-list">
          {joinedMissions
            .filter((mission) => mission.reserved === true)
            .map((mission) => (
              <li className="mission-list-item" key={mission.id}>
                {mission.mission_name}
              </li>
            ))}
        </ul>
      </section>
      <section className="reserved-rockets">
        <h1>My Rockets</h1>
        <ul className="rocket-list">
          {rocketsData
            .filter((rocket) => rocket.reserved === true)
            .map((rocket) => (
              <li className="rocket-list-item" key={rocket.id}>
                {rocket.rocket_name}
              </li>
            ))}
        </ul>
      </section>
    </section>
  );
};

export default MyProfile;
