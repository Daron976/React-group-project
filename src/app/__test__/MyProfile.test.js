import renderer from 'react-test-renderer';
import MyProfile from '../__mocks__/MyProfile';

describe('Profile Items', () => {
  test('Profile container items renders as expected', async () => {
    const snapMissionItems = renderer.create(
      <MyProfile />,
    )
      .toJSON();
    expect(snapMissionItems).toMatchSnapshot();
  });
});
