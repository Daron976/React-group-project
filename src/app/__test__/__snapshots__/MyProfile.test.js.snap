// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile container items renders as expected 1`] = `
<section
  className="profile-container"
>
  <section
    className="join-missions"
  >
    <h1>
      My Missions
    </h1>
    <ul
      className="mission-list"
    >
      <li
        className="mission-list-item"
      />
    </ul>
  </section>
  <section
    className="reserved-rockets"
  >
    <h1>
      My Rockets
    </h1>
    <ul
      className="rocket-list"
    >
      <li
        className="rocket-list-item"
      />
    </ul>
  </section>
</section>
`;
